test/e2e/specs/editor/various/keep-styles-on-block-transforms.spec.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Keep styles on block transforms', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'Should keep colors during a transform', async ( {
+		page,
+		editor,
+	} ) => {
+		await page.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( '## Heading' );
+
+		await editor.openDocumentSettingsSidebar();
+		await page.click( 'role=button[name="Text"i]' );
+
+		await page.click( 'role=button[name="Color: Luminous vivid orange"i]' );
+
+		await page.click( 'role=button[name="Heading"i]' );
+		await page.click( 'role=menuitem[name="Paragraph"i]' );
+
+		await expect.poll( editor.getBlocks ).toMatchObject( [
+			{
+				name: 'core/paragraph',
+				attributes: { content: 'Heading' },
+			},
+		] );
+	} );
+
+	test( 'Should keep the font size during a transform from multiple blocks into multiple blocks', async ( {
+		page,
+		pageUtils,
+		editor,
+	} ) => {
+		// Create a paragraph block with some content.
+		await page.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( 'Line 1 to be made large' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'Line 2 to be made large' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'Line 3 to be made large' );
+		await pageUtils.pressKeys( 'shift+ArrowUp' );
+		await pageUtils.pressKeys( 'shift+ArrowUp' );
+		await page.click( 'role=radio[name="Large"i]' );
+		await page.click( 'role=button[name="Paragraph"i]' );
+		await page.click( 'role=menuitem[name="Group"i]' );
```

Comment:
Why are we transforming it to `Group` while the original test transformed it to `Heading`?

Reply:
I think we should follow what the original test was testing in case we unexpectedly lose coverage. Especially since the title of the test case suggests that we should compare the size of the transformed blocks.

Reply:
Then yes its as per original test case only, let me know if any further feedbacks!

Reply:
Hmm, I think there might be some misunderstandings. I mean we should follow the original test case in https://github.com/WordPress/gutenberg/pull/49719/files#diff-0fbc4089adfabb18dad82dcdfbe924048b00dbdbab026fcdb9533ffacc4506baL50 and transform it to `Heading` block, not `Group` block as shown here.

Reply:
Hi @kevin940726,

Sorry for the confusion!

I have updated the test case, please take a look!

Thanks!